tests/integration/pilot/dns_auto_allocation_test.go

```diff
@@ -0,0 +1,104 @@
+//go:build integ
+// +build integ
+
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pilot
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/check"
+	"istio.io/istio/pkg/test/framework/components/echo/deployment"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/util/retry"
+)
+
+func TestDNSAutoAllocation(t *testing.T) {
+	framework.
+		NewTest(t).
+		Run(func(t framework.TestContext) {
+			ns := namespace.NewOrFail(t, t, namespace.Config{
+				Prefix: "dns-auto-allocation",
+				Inject: true,
+			})
+			cfg := `apiVersion: networking.istio.io/v1beta1
+kind: ProxyConfig
+metadata:
+  name: enable-dns-auto-allocation
+spec:
+  selector:
+    matchLabels:
+      app: a
+  environmentVariables:
+    ISTIO_META_DNS_CAPTURE: "true"
```

Comment:
nit: this shouldn't be needed, the test already has dns allocation enabled globally